src/hotplug_libusb.c

```diff
@@ -769,16 +776,21 @@ static LONG HPRemoveHotPluggable(int reader_index)
 
 	RFRemoveReader(readerTracker[reader_index].fullName,
 		PCSCLITE_HP_BASE_PORT + reader_index, REMOVE_READER_FLAG_REMOVED);
-	free(readerTracker[reader_index].fullName);
-	readerTracker[reader_index].status = READER_ABSENT;
-	readerTracker[reader_index].bus_device[0] = '\0';
-	readerTracker[reader_index].fullName = NULL;
+	HPCleanupHotPluggable(reader_index);
 
 	pthread_mutex_unlock(&usbNotifierMutex);
 
 	return 1;
 }	/* End of function */
 
+static void HPCleanupHotPluggable(int reader_index)
+{
+	free(readerTracker[reader_index].fullName);
```

Comment:
I'm unsure whether `usbNotifierMutex` needs to be taken here. Supposedly it's not needed on shutdown, but the threading model is not totally clear to me.